code/modules/admin/medal_panel/medals_panel_tgui.dm

```diff
@@ -0,0 +1,77 @@
+GLOBAL_DATUM_INIT(medals_panel, /datum/medals_panel_tgui, new)
+
+/datum/medals_panel_tgui
+	var/name = "Medals Panel"
+	var/starting_page = MEDALS_PANEL_XENO
+
+/datum/medals_panel_tgui/tgui_interact(mob/user, datum/tgui/ui, var/selected_panel = MEDALS_PANEL_XENO)
+	if(selected_panel == MEDALS_PANEL_USCM || selected_panel == MEDALS_PANEL_XENO)
+		starting_page = selected_panel
+
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "MedalsPanel", "Medals Panel")
+		ui.open()
+		ui.set_autoupdate(FALSE)
```

Comment:
this is a globally shared datum, do not have a 'starting_page' variable that changes whenever someone interacts with this, please.

Reply:
use ui.send_update instead, include custom data

Reply:
okay, I'd just get rid of the selected_panel param and forget about passing it in, since it's not a good idea to be changing a variable like that and passing it in as static data.